src/neo/SmartContract/Contract.cs

```diff
@@ -75,13 +75,14 @@ public static byte[] CreateMultiSigRedeemScript(int m, params ECPoint[] publicKe
                 throw new ArgumentException();
             using (ScriptBuilder sb = new ScriptBuilder())
             {
-                sb.EmitPush(m);
+                sb.EmitPush(m); // Signatures
+                sb.EmitPush(m); // m
```

Comment:
It's not worth changing a script, the script has all the data needed, no more, no less, `m` is there. I think the developer already declared his intention for `m` when he called `CreateMultiSigRedeemScript` with appropriate parameter.

Reply:
For wallets it's safe, if you call the syscall directly you need to check first the signatures, I think that it can produce some issues with inexperienced neo developers.

Reply:
@devhawk what do you think?

Reply:
As I understand the issue @shargon opened, if a contract is expecting 3 of 4 signatures and the developer doesn't check the signature count and passes in 1 or 2 signatures to `CheckMultiSig`, it would erroneously return true. I can see why @shargon is worried about this.

On the other hand, it seems reasonable that a developer using `CheckMultiSig` should understand how many valid signatures there should be. If the developer doesn't understand valid signatures count, they probably won't pass the correct value to `CheckMultiSig` anyway.

I'm with @roman-khimov on this. adding the m parameter to `CheckMultiSig` so we can add a `if (signatures.Length < m) return false;` call in `CheckMultiSigWithECDsa` doesn't seem worth it.